mlflow/store/artifact/gcs_artifact_repo.py

```diff
@@ -103,9 +132,10 @@ def log_artifacts(self, local_dir, artifact_path=None):
                 upload_path = posixpath.join(dest_path, rel_path)
             for f in filenames:
                 path = posixpath.join(upload_path, f)
-                gcs_bucket.blob(path, chunk_size=self._GCS_UPLOAD_CHUNK_SIZE).upload_from_filename(
-                    os.path.join(root, f), timeout=self._GCS_DEFAULT_TIMEOUT
-                )
+                # For large models, we need to speculatively retry a credential refresh
+                # and throw if it still fails.  We cannot use the built-in refresh because UC
+                # does not return a refresh token with the oauth token
+                self._retryable_log_artifact(bucket, path, os.path.join(root, f))
```

Comment:
What's preventing us from using [_retry_with_new_creds](https://github.com/mlflow/mlflow/blob/ee4fae72121d75848d729aa3b71b2036800126a3/mlflow/store/artifact/cloud_artifact_repo.py#L88-L103) here? We can move that util function from `cloud_artifact_repo` up to `artifact_repo` so it's accessible. 
Looking at [optimized_s3_artifact_repo](https://github.com/mlflow/mlflow/blob/6814e5a6556ebf957e5f4381e3d548415f77495a/mlflow/store/artifact/optimized_s3_artifact_repo.py#L207-L227) as an example, let's make
```
def try_func(gcs_bucket):
  gcs_bucket.blob(path, chunk_size=self._GCS_UPLOAD_CHUNK_SIZE).upload_from_filename(
     file_name, timeout=self._GCS_DEFAULT_TIMEOUT
  )

def creds_func(): 
  from google.cloud.storage import Client
  from google.oauth2.credentials import Credentials

   new_token = self.credential_refresh_def()
   credentials = Credentials(new_token)
   self.client = Client(project="mlflow", credentials=credentials)
   return self._get_bucket(bucket)
```
and `og_creds = gcs_bucket`. This way we maintain the pattern with other artifact repos.

Reply:
So the issue im running into with that is that the other retry's only need to accept the creds, but this one will need the bucket, the path, and the filename.

Reply:
You could pass all these arguments in as a tuple or a dict and retrieve them in the `try_func`.